src/s3/commands/downloadFileAs.ts

```diff
@@ -136,11 +138,17 @@ export async function downloadFileAsCommand(
             }
         )
 
+        setDefaultDownloadPath(saveLocation.fsPath)
```

Comment:
This should happen immediately after `promptForSaveLocation()`. Otherwise, while a big file is downloading the path still isn't saved. Testing the feature confirms that this doesn't feel like the expected behavior.